ratelimit/redis.go

```diff
@@ -121,17 +117,27 @@ func (c *clusterLimitRedis) startSpan(ctx context.Context, spanName string) func
 func (c *clusterLimitRedis) AllowContext(ctx context.Context, clearText string) bool {
 	c.metrics.IncCounter(redisMetricsPrefix + "total")
 	now := time.Now()
-	finishSpan := c.startSpan(ctx, allowSpanName)
+
+	var span opentracing.Span
+	parentSpan, ok := parentSpan(ctx)
+	if ok {
+		span = c.ringClient.StartSpan(allowSpanName, opentracing.ChildOf(parentSpan.Context()))
+	}
+	c.setCommonTags(span)
 
 	allow, err := c.allow(ctx, clearText)
 	failed := err != nil
-
-	finishSpan(failed)
-	c.measureQuery(allowMetricsFormat, allowMetricsFormatWithGroup, &failed, now)
-
 	if failed {
 		allow = !c.failClosed
+		setError(span, true)
+	}
+	if span != nil {
+		span.SetTag("allowed", allow)
+		span.Finish()
```

Comment:
Maybe it is better to do `defer span.Finish()` at creation site

Reply:
yes then we also have the same pattern in both places